roachtest: pass RunOptions to RunWithDetails(SingleNode)

herkolategan opened the pull request:
This change updates the `RunWithDetails` and `RunWithDetailsSingleNode` cluster interface methods to take `install.RunOptions` as a parameter instead of `option.NodeListOption` to expose the additional available options to `roachtest`.

Epic: None
Release Note: None

herkolategan committed:
roachtest: pass RunOptions to RunWithDetails(SingleNode)

This change updates the `RunWithDetails` and `RunWithDetailsSingleNode` cluster
interface methods to take `install.RunOptions` as a parameter instead of
`option.NodeListOption` to expose the additional available options to
`roachtest`.

Epic: None
Release Note: None

renatolabs commented on pkg/cmd/roachtest/cluster.go:
```diff
@@ -2321,25 +2321,37 @@ func (c *clusterImpl) RunE(ctx context.Context, options install.RunOptions, args
+	nodes := option.NodeListOption{}
+	for _, n := range options.Nodes {
+		nodes = append(nodes, int(n))
+	}
```
Heh, this kind of code is so silly, we should have used a single `[]int` custom type (at most) between roachprod and roachtest from the beginning 🤦.

We might be interested in having  a "conversion" function that does this for us, just like we have `InstallNodes()` to do the conversion in the opposite direction currently. Optional.

renatolabs reviewed (approved):
Nice!

herkolategan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/117613)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (waiting on @DarrylWong and @renatolabs)

___
*[`pkg/cmd/roachtest/cluster.go` line 2330 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/117613#gh-1453638201:-NoIL8NU8X4rDU-CGLoY:blrnnte) ([raw file](https://github.com/cockroachdb/cockroach/blob/f39e5c09db740d6b4a04ffad6931530c3eb2c628/pkg/cmd/roachtest/cluster.go#L2330)):*
<details><summary><i>Previously, renatolabs (Renato Costa) wrote…</i></summary><blockquote>

Heh, this kind of code is so silly, we should have used a single `[]int` custom type (at most) between roachprod and roachtest from the beginning 🤦.

We might be interested in having  a "conversion" function that does this for us, just like we have `InstallNodes()` to do the conversion in the opposite direction currently. Optional.
</blockquote></details>

Agreed, the thought to have a single type also crossed my mind. I think I'll make an issue for it somewhere, bound to run into it again.


<!-- Sent from Reviewable.io -->

herkolategan committed:
roachtest: add FromInstallNodes convenience function

Previously, the run functions in `cluster.go` each converted from install.Nodes
to a NodeListOption. This change adds a new convenience function
FromInstallNodes to do the conversion.

Epic: None
Release Note: None

herkolategan commented:
TFTR!
bors r=renatolabs
